rotkehlchen/chain/ethereum/modules/eigenlayer/balances.py

```diff
@@ -0,0 +1,125 @@
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+def _read_underlying_assets(
+        evm_inquirer: 'EthereumInquirer',
+        strategy_address: ChecksumEvmAddress,
+        depositor: ChecksumEvmAddress,
+) -> tuple['FVal', 'EvmToken']:
+    """
+    Query the amount deposited in a eigenlayer strategy and the token of the strategy
+    May raise:
+    - RemoteError
+    - NotERC20Conformant
+    """
+    contract = EvmContract(
+        address=strategy_address,
+        abi=UNDERLYING_BALANCES_ABI,
+        deployed_block=0,
+    )
+    deposited_amount_raw = contract.call(
+        node_inquirer=evm_inquirer,
+        method_name='userUnderlyingView',
+        arguments=[depositor],
+    )
+    underlying_token_address = contract.call(
+        node_inquirer=evm_inquirer,
+        method_name='underlyingToken',
+    )
+    token = get_or_create_evm_token(
+        userdb=evm_inquirer.database,
+        evm_address=underlying_token_address,
+        chain_id=evm_inquirer.chain_id,
+    )
+    amount = asset_normalized_value(
+        amount=deposited_amount_raw,
+        asset=token,
+    )
+    return amount, token
+
+
+class EigenlayerBalances(ProtocolWithBalance):
+    def __init__(
+            self,
+            database: DBHandler,
+            evm_inquirer: 'EthereumInquirer',
+            chain_id: 'CHAIN_IDS_WITH_BALANCE_PROTOCOLS',
+    ):
+        super().__init__(
+            database=database,
+            evm_inquirer=evm_inquirer,
+            chain_id=chain_id,
+            counterparty=CPT_EIGENLAYER,
+            deposit_event_types={(HistoryEventType.STAKING, HistoryEventSubType.DEPOSIT_ASSET)},
+        )
+        self.evm_inquirer: 'EthereumInquirer'
+        self.web3 = self.evm_inquirer.web3_mapping
+        self.inquirer = Inquirer()
+
+    def query_balances(self) -> 'BalancesType':
+        """Query underlying balances for deposits in eigenlayer"""
+        balances: BalancesType = defaultdict(lambda: defaultdict(Balance))
+        # fetch deposit events
+        addresses_with_deposits = self.addresses_with_deposits(products=[EvmProduct.STAKING])
+        # remap all events into a list that will contain all pairs (depositor, strategy)
+        deposits = set()
+        for depositor, event_list in addresses_with_deposits.items():
+            for event in event_list:
+                if event.extra_data is None:
+                    continue
+                if (strategy := event.extra_data.get('strategy')) is not None:
+                    deposits.add((depositor, strategy))
+
+        # user had no correctly processed events
```

Comment:
Is correctly processed the right thing to say here? Can it be they had no relevant events simply?

```suggestion
        # user had no related events
```